### ISSUE TITLE
Add HTML5 API Interfaces to Window

### ISSUE THREAD

**Issue by oleersoy** [microsoft/TypeScript#28525]
Attempting to define a check for HTML5 File API support in Angular like this:

```
private hasHtml5FileApiSupport;
  constructor(@Optional() @Inject(DOCUMENT) document: Document) {
    const w = document.defaultView;
    this.hasHtml5FileApiSupport = w.File && w.FileReader && w.FileList && w.Blob;
```

However VSCode draws red squigglies under `File, FileReader, and FileList` since there are not part of the Window interface.  It would be nice if they were.  We can still do NPE checks on them.

**Comment by oleersoy** [comment]
https://stackoverflow.com/questions/53292312/checking-for-html5-file-api-support-in-angular


https://stackoverflow.com/questions/53305340/typescript-html5-api-typings/53305917#53305917

**Comment by oleersoy** [comment]
Related Angular issue:
https://github.com/angular/angular/issues/7341

**Comment by jakebailey** [comment]
Just checking old issues; this one was fixed in #37464. [Playground Link](https://www.typescriptlang.org/play?ts=5.1.6#code/GYVwdgxgLglg9mABAIwDYEMAWAKAJnCEAWwFMwoAuRAEQOLKgEpEBvAWAChFvEIEBnKIgDuiALyJ8hUuQB0uEsHQhUUAGowSwgIQBuTj14ChAD3EjZAMRioSiAGT2L12wCUS6BQCcHT4VZsSABkYQV8LACFUOGR9DgBfIA)
